clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsResponse.java

```diff
@@ -152,19 +151,19 @@ public boolean zkMigrationReady() {
         return data.zkMigrationReady();
     }
 
-    public static ApiVersionsResponse parse(ByteBuffer buffer, short version) {
+    public static ApiVersionsResponse parse(ByteBufferAccessor readable, short version) {
```

Comment:
The rest of the implementations of `parse` but this one take a `Readable`. Did you consider adding `slice()` to `Readable`? If you do that, this implementation can call `slice()`so that it can parse the response twice if the version is incorrect.

Reply:
Done.